helix-core/src/main/java/org/apache/helix/model/ClusterConfig.java

```diff
@@ -160,7 +161,8 @@
       DEFAULT_GLOBAL_REBALANCE_PREFERENCE =
```

Comment:
I think the change does make sense. But is this value still being used anymore?

Maybe we can change the logic to, if nothing is configured, then apply this default value.
If partially missing, then put whatever user set in the ClusterConfig but leave the processing logic to the rebalancer like what you are doing right now.

Reply:
This default map is still being used, and the current behavior is exactly as you have described. Looks to me there's nothing to change here :)

Reply:
Yeah, with your revised version, it looks good now. Before, since you hardcoded the value there, it is not really using the default numbers.

There is one more thing I suggest to change. It is not used anymore in this file. So we shall move it to ConstraintBasedAlgorithmFactory.java. Since the default preference is no longer part of the config class logic anymore.

Reply:
It's still used by `getGlobalRebalancePreference()` in the config class. I think it's fine where it is, because it has the format of a map, and logically belongs to ClusterConfig. Technically its default values should belong in the factory class, but that may be an overkill.